### PR TITLE
Madina/enhancement/305 improve registration flow

### LINKED ISSUE #305: Improve Registration Flow
## 📝 **Task Description** 

https://github.com/Alt-Org/Altzone-WebPages/issues/291#issue-2678463154

> After 'Tili on luotu' popup I am stuck here in 'Rekisteröidy' page.
> IMO you should redirect on success to 'Login' page, or even better login automatically and continue from the page where this sequence was originally started from.
> 
> If I press 'Lähetä' again, something bad happens in your client as it seems not to be able to handle user error correctly.

---

## ✅ **Steps to Complete**

Enhance the registration process to improve user experience and error handling:

1. Automatically log the user in upon successful registration and redirect them to the homepage.
2. Handle repeated submissions (e.g., pressing 'Lähetä' again) and manage 409 conflict errors gracefully by notifying the user of the issue via a popup or appropriate feedback.
3. Ensure the redirect logic is robust, even if a failure occurs.
4. Test the updated registration flow to confirm it functions correctly and handles all edge cases.
5. Update or add automated tests as necessary to cover the new logic.

---

### PR DESCRIPTION
**Pull request by madinamacow**
closes #305 

## 🔧 **Changes Made**

1. **CustomForm.module.scss**
    - Reduced form padding
    - Increased error message font size for better readability
2. **useRegisterForm.ts**
    - Added automatic login and homepage redirect after successful registration
    - Added user feedback for duplicate submissions and 409 errors
    - Improved redirect handling reliability
    - Tested all flows and edge cases
    - Updated test suite for new scenarios

### Summary

These updates enhance the registration experience by implementing automatic login with homepage redirect upon success. The changes include improved error handling with clear feedback for duplicate submissions and 409 errors. All flows have been thoroughly tested, and the automated test suite has been expanded. Error messages are now more readable thanks to increased font size, particularly benefiting mobile users.

---

- **Functionality**: I have tested my code, and it works as expected.

### CODE REVIEW

File: frontend-next-migration/src/app/_styles/variables/global.scss [hunk 1]
```diff
@@ -7,9 +7,9 @@
   --font-size-s: 14px;
   --font-line-s: 20px;
   --font-s: var(--font-size-s) / var(--font-line-s) var(--font-family-main);
-  //--font-s: var(--font-size-s) / var(--font-line-s);
+  // --font-s: var(--font-size-s) / var(--font-line-s);
 
-  --font-size-xs: 12px;
+  --font-size-xs: 17px;
```
Comment (Skoivumaki) [inline frontend-next-migration/src/app/_styles/variables/global.scss:12]: You are changing global variables, are you absolutely sure this change is not affecting something else and potentially breaking it? Anyway if the error text size is too small, which it definetly is. Change the font to --font-m or similar in `CustomForm.module.scss` no need to change global variables.

File: frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts [hunk 2]
```diff
@@ -1,66 +1,95 @@
...
+                }),
+            );
+
+            // Redirect to home page
+            router.push('/');
+        } catch (error: any) {
+            const errorMessage =
+                error?.data?.message?.[0] ??
+                error?.data?.message ??
+                t('Käyttäjänimi on jo käytössä, valitse toinen käyttäjänimi');
```
Comment (Skoivumaki) [inline frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts:80]: Remember to add i18n.

File: frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts [hunk 3]
```diff
@@ -1,66 +1,95 @@
...
-        const ObjectToBeSent: IUserRegisterDto = {
-            username: fieldValues.username,
-            password: fieldValues.password,
-            repeatPassword: fieldValues.password,
-            Player: {
-                uniqueIdentifier: fieldValues.username,
-                backpackCapacity: 100,
-                name: fieldValues.username,
-                above13: fieldValues.ageConsent,
-                parentalAuth: false,
```
Comment (Skoivumaki) [inline frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts:40]: Why change `parentalAuth` to true?

**Review by Skoivumaki — CHANGES_REQUESTED** [review]
Build and updated tests pass. Redirecting after successful registration seems to work good. Check comments.

File: frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts (hunk 3 above)
Comment (madinamacow) [reply frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts:40]: I accidentally mixed it with the test file. That wasn't my intention.

Comment (Skoivumaki) [reply frontend-next-migration/src/features/AuthByUsername/model/useRegisterForm.ts:40]: No worries, I was just wondering if you knew something I didnt. I did some testing and parentalAuth being declared isnt actually required anymore, but it may stay for the future.

**Review by Jonroi — APPROVED** [review]
All previous comments have been fixed, and everything is working and looking as planned—great job!
